src/class-rest-api-endpoints.php

```diff
@@ -138,20 +148,26 @@ public function get_footer_settings(): WP_Error|array {
 	 *
 	 * @param WP_REST_Request $request The request object.
 	 *
-	 * @return array
+	 * @return array{
+	 *   Status?: string,
+	 * }  The newsletter status.
 	 */
 	public function get_status( WP_REST_Request $request ): array {
 		$post_id = $request->get_param( 'post_id' );
-		if ( empty( $post_id ) ) {
+		if ( empty( $post_id ) || ! is_int( $post_id ) ) {
```

Comment:
Is there any time this might be a string '1234' in which case we could use absint() maybe? not sure.

Reply:
This was a great catch. Thank you. It actually would _always_ be a string, so adding this `is_int()` check actually would break this method without `absint()`! I've fixed it. 🙌